File: src/main/java/io/polygloat/dtos/request/EditRepositoryDTO.java
package io.polygloat.dtos.request;

import io.polygloat.dtos.request.validators.annotations.RepositoryRequest;

import javax.validation.constraints.NotNull;

@RepositoryRequest
public class EditRepositoryDTO extends AbstractRepositoryDTO {
    @NotNull
    Long repositoryId;


    public EditRepositoryDTO(Long repositoryId, String name) {
        this.repositoryId = repositoryId;
        this.name = name;
    }

    public EditRepositoryDTO() {
    }

    public @NotNull Long getRepositoryId() {
        return this.repositoryId;
    }

    public void setRepositoryId(@NotNull Long repositoryId) {
        this.repositoryId = repositoryId;
    }
}

File: src/main/java/io/tolgee/dtos/request/EditRepositoryDTO.kt
package io.tolgee.dtos.request

import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

data class EditRepositoryDTO(
        @field:NotNull
        var projectId: Long? = null,

        @field:NotNull @field:Size(min = 3, max = 500)
        var name: String? = null,

        @field:Size(min = 3, max = 60)
        @field:Pattern(regexp = "^[a-z0-9]*[a-z]+[a-z0-9]*$", message = "invalid_pattern")
        var addressPart: String? = null
)
